speed-dreams/source-2.2.3/data/data/menu/ResearcherMenu.xml

```diff
@@ -6,31 +6,21 @@
 
     <section name="dynamic controls">
 
-        <!-- Checkboxes for selecting a Task -->
-        <section name="CheckboxTaskLaneKeeping">
-            <attstr name="type" val="check box"/>
-            <attstr name="text" val="Lane keeping"/>
-            <attstr name="tip" val="Select the Lane Keeping task"/>
+        <!-- RadioBoxList for selecting a Task -->
+        <section name="TaskRadioButtons">
+            <attstr name="type" val="radio buttons"/>
+            <attstr name="text" val="Undefined Task"/> <!-- Keep length high for overwriting -->
+            <attstr name="tip" val="Select the task"/>
             <attnum name="x" val="20"/>
             <attnum name="y" val="370"/>
             <attnum name="image width" val="20"/>
             <attnum name="image height" val="20"/>
             <attstr name="color" val="0xFFFFFF"/>
             <attstr name="font" val="small_t"/>
-            <attstr name="checked" val="no"/>
-        </section>
-
-        <section name="CheckboxTaskSpeedControl">
-            <attstr name="type" val="check box"/>
-            <attstr name="text" val="Speed control"/>
-            <attstr name="tip" val="Select the Speed Control task"/>
-            <attnum name="x" val="20"/>
-            <attnum name="y" val="340"/>
-            <attnum name="image width" val="20"/>
-            <attnum name="image height" val="20"/>
-            <attstr name="color" val="0xFFFFFF"/>
-            <attstr name="font" val="small_t"/>
-            <attstr name="checked" val="no"/>
+            <attnum name="selected" val="0"/>
+            <attnum name="amount" val="2"/>
```

Comment:
The amount is now specified in the xml, but the actual values are specified in the code. I think it would be best to move everything to the xml. So instead of 'amount' we would have a section 'options' or something like that, which specifies all the options. There is already a function called `GfParmGetEltNb` which returns you the amount of childs of a section.

The create control function is then also simplified, because it can pull all the data from xml instead of having to create a list of options there in the code and giving that as parameter.